Tests/Runtime/MediaStreamTest.cs

```diff
@@ -1,101 +1,201 @@
+            var op5 = peer2.SetLocalDescription(ref op4.desc);
+            yield return op5;
+            var op6 = peer1.SetRemoteDescription(ref op4.desc);
+            yield return op6;
+
+            var op7 = new WaitUntilWithTimeout(() =>
+                peer1.IceConnectionState == RTCIceConnectionState.Connected ||
+                peer1.IceConnectionState == RTCIceConnectionState.Completed, 5000);
+            yield return op7;
+            Assert.True(op7.IsCompleted);
+
+            var op8 = new WaitUntilWithTimeout(() =>
+                peer2.IceConnectionState == RTCIceConnectionState.Connected ||
+                peer2.IceConnectionState == RTCIceConnectionState.Completed, 5000);
+            yield return op8;
+            Assert.True(op7.IsCompleted);
+
+            var op9 = new WaitUntilWithTimeout(() => pc2Senders.Count > 0, 5000);
+            yield return op9;
+            Assert.True(op9.IsCompleted);
+
+            foreach (var sender in pc1Senders)
+            {
+                peer1.RemoveTrack(sender);
+            }
+
+            foreach (var sender in pc2Senders)
+            {
+                peer2.RemoveTrack(sender);
+            }
+            pc1Senders.Clear();
+            peer1.Close();
+            peer2.Close();
+            audioStream.Dispose();
         }
-        pc1Senders.Clear();
-        GameObject.DestroyImmediate(camObj);
 
-        peer1.Close();
-        peer2.Close();
+        [UnityTest]
+        [Timeout(5000)]
+        [UnityPlatform(exclude = new[] {RuntimePlatform.OSXEditor})] // TODO:: fix Crush bug on Yamato
+        public IEnumerator MediaStreamTest_AddAndRemoveVideoTrack()
+        {
+            var camObj = new GameObject("Camera");
+            var cam = camObj.AddComponent<Camera>();
+            RTCConfiguration config = default;
+            config.iceServers = new[]
+            {
+                new RTCIceServer {urls = new[] {"stun:stun.l.google.com:19302"}}
+            };
+
+            var pc1Senders = new List<RTCRtpSender>();
+            var pc2Senders = new List<RTCRtpSender>();
+            var peer1 = new RTCPeerConnection(ref config);
+            var peer2 = new RTCPeerConnection(ref config);
+
+            peer1.OnIceCandidate = candidate => { peer2.AddIceCandidate(ref candidate); };
+            peer2.OnIceCandidate = candidate => { peer1.AddIceCandidate(ref candidate); };
+            peer2.OnTrack = e => { pc2Senders.Add(peer2.AddTrack(e.Track)); };
+            var videoStream = cam.CaptureStream(1280, 720);
+            yield return new WaitForSeconds(1.0f);
+
+            foreach (var track in videoStream.GetTracks())
+            {
+                pc1Senders.Add(peer1.AddTrack(track));
+            }
 
-        yield return videoStream.FinalizeEncoder();
-        videoStream.Dispose();
+            RTCOfferOptions options1 = default;
+            RTCAnswerOptions options2 = default;
+            var op1 = peer1.CreateOffer(ref options1);
```

Comment:
Can we make a function to share some code with MediaStreamTest_AddAndRemoveAudioTrack() ?